pkg/machine/wsl/util_windows.go

```diff
@@ -98,37 +95,7 @@ func winVersionAtLeast(major uint, minor uint, build uint) bool {
 }
 
 func HasAdminRights() bool {
-	var sid *windows.SID
-
-	// See: https://coolaj86.com/articles/golang-and-windows-and-admins-oh-my/
-	if err := windows.AllocateAndInitializeSid(
-		&windows.SECURITY_NT_AUTHORITY,
-		2,
-		windows.SECURITY_BUILTIN_DOMAIN_RID,
-		windows.DOMAIN_ALIAS_RID_ADMINS,
-		0, 0, 0, 0, 0, 0,
-		&sid); err != nil {
-		logrus.Warnf("SID allocation error: %s", err)
-		return false
-	}
-	defer func() {
-		_ = windows.FreeSid(sid)
-	}()
-
-	//  From MS docs:
-	// "If TokenHandle is NULL, CheckTokenMembership uses the impersonation
-	//  token of the calling thread. If the thread is not impersonating,
-	//  the function duplicates the thread's primary token to create an
-	//  impersonation token."
-	token := windows.Token(0)
-
-	member, err := token.IsMember(sid)
-	if err != nil {
-		logrus.Warnf("Token Membership Error: %s", err)
-		return false
-	}
-
-	return member || token.IsElevated()
+	return windows.GetCurrentProcessToken().IsElevated()
```

Comment:
`golang.org/x/sys/windows` now supports a function to directly determine if the current process has administrator privileges.

Ref: https://github.com/golang/sys/commit/61b9204099cb1bebc803c9ffb9b2d3acd9d457d9 / https://go-review.googlesource.com/c/sys/+/176858

> BTW. k8s has already done this: https://github.com/kubernetes/kubernetes/pull/108146

Reply:
Do all golang versions support this fflavor of windows? if so, I see no objection.

Reply:
• The `IsElevated` API has been supported since the first version of `github.com/golang/sys`
• The Windows API used internally by the `IsElevated` function supports a minimum of _Windows XP_

Reply:
These are not equivalent operations there is two parts to this operation. It's checking if the token is a member of the admin group, or if the process itself was marked as elevated. The former is the part that is most common the latter doesnt matter as much.

Reply:
For the HasAdminRights function itself, it is equivalent. IsElevated corresponds to the two situations you just mentioned. For the functionality of the HasAdminRights function, we do not need to make a more detailed distinction here.

Reply:
Relying on elevated split tokens might be reliable in recent versions of Windows but it wasn't at the time this was written. See linked article in the code. Before dropping it would need to be checked against all editions of server, 10, and 11.